RedHat-UX/red-hat-design-system
# CI/CD

## What I did

1. add changesets in prerelease mode with the `beta` version suffix
2. add ci/cd via web-test-runner
3. add an esbuild build script
4. add package exports, so that one could
    ```js
    // good
    import '@rhds/elements/rh-alert/rh-alert.js';
    // bad
    import '@rhds/elements/elements/rh-alert/rh-alert.js';
    ```
    and get the file located at `/elements/rh-alert/rh-alert.js`

## Testing Instructions

1. `npm pack` to verify the npm tarball has all the right stuff and none of the wrong stuff
2. try unpacking that tarball into a fresh npm project, and see if you can 
    ```js
    import '@rhds/elements/rh-alert/rh-alert.js';
    ```
3. see if CI passes / why it fails

## Notes to Reviewers
We can merge once review passes, but please don't add a changeset for `rh-alert` until #265 is resolved, thanks

Hope you're having a nice day :)
